ImprovedAnvil/lib/optional/all_protagonists_extra_rewards.TPH

```diff
@@ -0,0 +1,36 @@
+// Remove class restrictions from nonessential kiit drios
+
+// Golden Girdle in Watcher's Keep level 4
+COPY_EXISTING ~override\AR3017.BCS~ ~override\AR3017.BCS~
+  DECOMPILE_AND_PATCH BEGIN
+    REPLACE_TEXTUALLY ~!Class(Player1,MAGE_ALL)~ ~~
+  END
+
+// Scroll of Memory Boosting in Spellhold
+COPY_EXISTING ~override\AR154.BCS~ ~override\AR1514.BCS~
+  DECOMPILE_AND_PATCH BEGIN
+    REPLACE_TEXTUALLY ~IF Global("Iaitem","AR1514",0) Global("closedDoor","AR1514",2) OR(2) Class(Player1,MAGE_ALL) Class(Player1,DRUID_ALL) !Class(Player1,FIGHTER_ALL) !Class(Player1,THIEF_ALL) !Class(Player1,CLERIC_ALL) THEN RESPONSE #100 SetGlobal("Iaitem","AR1514",1) GiveItemCreate("S!misc01",Player1,0,0,0) END~ ~IF Global("Iaitem1","AR1514",0) Global("closedDoor","AR1514",2) THEN RESPONSE #100 SetGlobal("Iaitem1","AR1514",1) GiveItemCreate("S!misc01",Player1,0,0,0) END~ 
+  END
+
+// Scrolls in Planar Sphere
+COPY_EXISTING ~override\AR0410.BCS~ ~override\AR0410.BCS~
```

Comment:
this is actually more complicated than you think
I've made a patch a year ago that made the scrolls add up continuously without the need for the player to come into the sphere after each batch in order to reset the counter. I've had to add patching of BALDUR.BCS as well.

Reply:
good point, i didn't patch baldur.bcs and it looks like there is a ranger quest (684) line as well as the mage scroll patch there (1007).